sys/build.rs

```diff
@@ -101,16 +184,50 @@ fn main() {
         patch(out_dir, patches_dir.join(file));
     }
 
+    let mut add_cflags = vec![];
+    if env::var("CARGO_CFG_TARGET_OS").unwrap() == "wasi" {
+        let wasi_sdk_path = get_wasi_sdk_path();
+        if !wasi_sdk_path.try_exists().unwrap() {
+            panic!(
+                "wasi-sdk not installed in specified path of {}",
+                wasi_sdk_path.display()
+            );
+        }
+        env::set_var("CC", wasi_sdk_path.join("bin/clang").to_str().unwrap());
+        env::set_var("AR", wasi_sdk_path.join("bin/ar").to_str().unwrap());
+        let sysroot = format!(
+            "--sysroot={}",
+            wasi_sdk_path.join("share/wasi-sysroot").display()
+        );
+        env::set_var("CFLAGS", &sysroot);
+        add_cflags.push(sysroot);
+    }
+
     // generating bindings
-    bindgen(out_dir, out_dir.join("quickjs.bind.h"), &defines);
+    bindgen(
+        out_dir,
+        out_dir.join("quickjs.bind.h"),
+        &defines,
+        add_cflags,
+    );
 
     let mut builder = cc::Build::new();
     builder
         .extra_warnings(false)
+        .flag("-Wno-implicit-const-int-float-conversion")
```

Comment:
This should probably be `flag_if_supported`. This flag doesn't exist on windows and cause the build to fail.

Reply:
fixed